.github/actions/gradle-task/action.yml

```diff
@@ -149,6 +152,6 @@ runs:
       if: failure() && inputs.failure-path-upload != 'null'
       uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4
       with:
-        name: specified-upload
+        name: ${{inputs.failure-upload-name}}
```

Comment:
@zach-klippenstein I also fixed test report upload on failure - so you can download the typical `.html` report and see the reasons for all your different test failures in CI.